diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerInterface.java b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerInterface.java
new file mode 100644
index 0000000..09d1e31
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerInterface.java
@@ -0,0 +1,11 @@
+package expo.modules.interfaces.barcodescanner;
+
+import android.graphics.Bitmap;
+
+import java.util.List;
+
+public interface BarCodeScannerInterface {
+  BarCodeScannerResult scan(byte[] imageData, int width, int height, int rotation);
+  List<BarCodeScannerResult> scanMultiple(Bitmap bitmap);
+  void setSettings(BarCodeScannerSettings settings);
+}
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerProviderInterface.java b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerProviderInterface.java
new file mode 100644
index 0000000..705037d
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerProviderInterface.java
@@ -0,0 +1,7 @@
+package expo.modules.interfaces.barcodescanner;
+
+import android.content.Context;
+
+public interface BarCodeScannerProviderInterface {
+  BarCodeScannerInterface createBarCodeDetectorWithContext(Context context);
+}
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerResult.java b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerResult.java
new file mode 100644
index 0000000..c2b9a3a
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerResult.java
@@ -0,0 +1,106 @@
+package expo.modules.interfaces.barcodescanner;
+
+import java.util.List;
+
+public class BarCodeScannerResult {
+  public static class BoundingBox {
+    private final int x;
+    private final int y;
+    private final int width;
+    private final int height;
+
+    public BoundingBox(int x, int y, int width, int height) {
+      this.x = x;
+      this.y = y;
+      this.width = width;
+      this.height = height;
+    }
+
+    public int getX() {
+      return x;
+    }
+
+    public int getY() {
+      return y;
+    }
+
+    public int getWidth() {
+      return width;
+    }
+
+    public int getHeight() {
+      return height;
+    }
+  }
+
+  private int mReferenceImageWidth;
+  private int mReferenceImageHeight;
+  private int mType;
+  private String mValue;
+  private String mRaw;
+  private List<Integer> mCornerPoints;
+
+
+  public BarCodeScannerResult(int type, String value, String raw, List<Integer> cornerPoints, int height, int width) {
+    mType = type;
+    mValue = value;
+    mRaw = raw;
+    mCornerPoints = cornerPoints;
+    mReferenceImageHeight = height;
+    mReferenceImageWidth = width;
+  }
+
+  public int getType() {
+    return mType;
+  }
+
+  public String getValue() { return mValue; }
+
+  public String getRaw() { return mRaw; }
+
+  public List<Integer> getCornerPoints() {
+    return mCornerPoints;
+  }
+
+  public void setCornerPoints(List<Integer> points) {
+    mCornerPoints = points;
+  }
+
+  public int getReferenceImageHeight() {
+    return mReferenceImageHeight;
+  }
+
+  public void setReferenceImageHeight(int height) {
+    mReferenceImageHeight = height;
+  }
+
+  public int getReferenceImageWidth() {
+    return mReferenceImageWidth;
+  }
+
+  public void setReferenceImageWidth(int width) {
+    mReferenceImageWidth = width;
+  }
+
+  public BoundingBox getBoundingBox() {
+    if (mCornerPoints.isEmpty()) {
+      return new BoundingBox(0, 0, 0, 0);
+    }
+    int minX = Integer.MAX_VALUE;
+    int minY = Integer.MAX_VALUE;
+    int maxX = Integer.MIN_VALUE;
+    int maxY = Integer.MIN_VALUE;
+
+    for (int i = 0; i < mCornerPoints.size(); i += 2) {
+      int x = mCornerPoints.get(i);
+      int y = mCornerPoints.get(i + 1);
+
+      minX = Integer.min(minX, x);
+      minY = Integer.min(minY, y);
+      maxX = Integer.max(maxX, x);
+      maxY = Integer.max(maxY, y);
+    }
+
+    return new BoundingBox(minX, minY, maxX - minX, maxY - minY);
+  }
+}
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettings.java b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettings.java
new file mode 100644
index 0000000..e0529ea
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettings.java
@@ -0,0 +1,28 @@
+package expo.modules.interfaces.barcodescanner;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BarCodeScannerSettings extends HashMap<BarCodeScannerSettingsKey, Object> {
+
+  public BarCodeScannerSettings() {
+    super();
+  }
+  public BarCodeScannerSettings(Map<String, Object> settings) {
+    super();
+    for (Map.Entry<String, Object> entry: settings.entrySet()) {
+      BarCodeScannerSettingsKey key = BarCodeScannerSettingsKey.fromStringName(entry.getKey());
+      if (key != null) {
+        put(key, entry.getValue());
+      }
+    }
+  }
+
+  public void putTypes(Object types) {
+    put(BarCodeScannerSettingsKey.TYPES, types);
+  }
+
+  public Object getTypes() {
+    return get(BarCodeScannerSettingsKey.TYPES);
+  }
+}
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettingsKey.java b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettingsKey.java
new file mode 100644
index 0000000..26a5f92
--- /dev/null
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/interfaces/barcodescanner/BarCodeScannerSettingsKey.java
@@ -0,0 +1,24 @@
+package expo.modules.interfaces.barcodescanner;
+
+public enum BarCodeScannerSettingsKey {
+  TYPES("barCodeTypes");
+
+  private final String mName;
+
+  BarCodeScannerSettingsKey(String stringName) {
+    mName = stringName;
+  }
+
+  public static BarCodeScannerSettingsKey fromStringName(String name) {
+    for (BarCodeScannerSettingsKey key: BarCodeScannerSettingsKey.values()) {
+      if (key.getName().equalsIgnoreCase(name)) {
+        return key;
+      }
+    }
+    return null;
+  }
+
+  public String getName() {
+    return mName;
+  }
+}
